src/rgw/rgw_client_io_decoimpl.h

```diff
@@ -0,0 +1,442 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_RGW_CLIENT_IO_DECOIMPL_H
+#define CEPH_RGW_CLIENT_IO_DECOIMPL_H
+
+#include <type_traits>
+#include <string>
+
+#include <boost/optional.hpp>
+
+#include "rgw_common.h"
+#include "rgw_client_io.h"
+
+template <typename T>
+class RGWRestfulIOAccountingEngine : public RGWDecoratedRestfulIO<T>,
```

Comment:
At the moment we have `rgw::io::AccountingEngine` (the filter) and `rgw::io::Accounter` (the interface). I'm still looking for better names.